### DISCUSSION TITLE
why not use i32 replace int?

### DISCUSSION THREAD

**Discussion by Silentdoer** [vlang/v#15375] [General]
it's not consistent with other number type

**Comment by tzSharing** [comment]
@Silentdoer OCD may be unbearable :smile:

**Comment by nachoverdon** [comment]
I can use them interchangeably without a problem. I don't know if this should be the case tho.

```v
fn main() {
	mut test := int(1)
	mut test2 := i32(1)

	if test == test2 {
		println("they are equal")
	} else {
		println("they are different")
	}

	test = return_i32(5)
	test2 = return_int(5)

	if test == test2 {
		println("they are equal")
	} else {
		println("they are different")
	}
}

fn return_int(num int) int {
	return int(num)
}

fn return_i32(num int) i32 {
	return i32(num)
}
```

**Comment by Wajinn** [comment]
Kind of like it that way, where int or i32 can be used.  "int" is so common in people's mind, it's almost like a tradition.  Likely to get, "Where is int?" or "Save the int!".  So perhaps its for the best.

**Comment by runeimp** [comment]
In C `int` is system specific. V (and most other languages) being able to interoperate with C typically treat `int` the same way. On an 8-bit architecture it's 8 bits on a 64-bit architecture it's 64 bits. As I understand it the reasoning was that if `int` the most common type of all variables could grow or shrink depending on system word size the code would be more portable compiling to different architectures. Since back in time when most _users_ were typically also programmers the users of a program would understand the limitations based on system architecture and only specify sized integers when it was absolutely necessary for the code to be reliable. Say for instance in data structures or complex math.

Even today do you care most of the time if an integer is 32 bits or 64 bits? Unless you're dealing with code that handles very large integers it's unlikely to matter. And if it truly matters you may need to use a special math library to use complex 128-bit numbers as well. Only when dealing with embedded systems do we really need to worry about the size of an `int` only being 8 or 16 bits.
